File: jar-common/src/main/java/com/beancounter/common/contracts/TrnRequest.java
package com.beancounter.common.contracts;

import com.beancounter.common.input.TrnInput;
import java.util.Collection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TrnRequest implements Payload<Collection<TrnInput>> {

  private Collection<TrnInput> data;
  @NonNull
  private String portfolioId;
}

File: jar-common/src/main/kotlin/com/beancounter/common/contracts/TrnRequest.kt
package com.beancounter.common.contracts

import com.beancounter.common.input.TrnInput

/**
 * Request to import transactions for a portfolio.
 */
data class TrnRequest(
    val portfolioId: String,
    override val data: List<TrnInput>
) : Payload<List<TrnInput>>